lib/Sema/NameBinding.cpp

```diff
@@ -902,55 +917,71 @@ void NameBinder::crossImport(ModuleDecl *M, UnboundImport &I) {
     // FIXME: Should we warn if M doesn't reexport underlyingModule?
     SF.addSeparatelyImportedOverlay(M, I.getUnderlyingModule().get());
 
-  // FIXME: Most of the comparisons we do here are probably unnecessary. We
-  // only need to findCrossImports() on pairs where at least one of the two
-  // modules declares cross-imports, and most modules don't. This is low-hanging
-  // performance fruit.
-
   auto newImports = crossImportableModules.getArrayRef()
                         .slice(nextModuleToCrossImport);
```

Comment:
Not part of this PR but it would be better to use `drop_front` instead of `slice` as the latter's semantics are not clear from the name.